src/main/java/org/codehaus/plexus/archiver/zip/AbstractZipArchiver.java

```diff
@@ -506,22 +508,15 @@ protected void zipFile( final ArchiveEntry entry, ConcurrentJarCreator zOut, Str
 
         final boolean b = entry.getResource() instanceof SymlinkDestinationSupplier;
         String symlinkTarget = b ? ( (SymlinkDestinationSupplier) entry.getResource() ).getSymlinkDestination() : null;
-        InputStreamSupplier in = new InputStreamSupplier()
-        {
-
-            @Override
-            public InputStream get()
+        InputStreamSupplier in = () -> {
+            try
             {
-                try
-                {
-                    return entry.getInputStream();
-                }
-                catch ( IOException e )
-                {
-                    throw new RuntimeException( e );
-                }
+                return entry.getInputStream();
+            }
+            catch ( IOException e )
+            {
+                throw new UncheckedIOException( e );
```

Comment:
I didn't knew about this new exception. Looks really useful for lambdas. Thanks for sharing it.